src/main/java/net/neoforged/neoforge/client/entity/animation/json/AnimationLoader.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.client.entity.animation.json;
+
+import com.google.common.collect.MapMaker;
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParseException;
+import com.mojang.logging.LogUtils;
+import com.mojang.serialization.JsonOps;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import net.minecraft.client.animation.AnimationDefinition;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.server.packs.resources.ResourceManager;
+import net.minecraft.server.packs.resources.SimpleJsonResourceReloadListener;
+import net.minecraft.util.profiling.ProfilerFiller;
+import org.jetbrains.annotations.Nullable;
+import org.slf4j.Logger;
+
+/**
+ * A loader for entity animations written in JSON. You can also get parsed animations from this class.
+ */
+public final class AnimationLoader extends SimpleJsonResourceReloadListener {
+    private static final Logger LOGGER = LogUtils.getLogger();
+
+    public static final AnimationLoader INSTANCE = new AnimationLoader();
+
+    private final Map<ResourceLocation, AnimationHolder> animations = new MapMaker().weakValues().concurrencyLevel(1).makeMap();
+    @SuppressWarnings("MismatchedQueryAndUpdateOfCollection")
+    private final List<AnimationHolder> strongHolderList = new ArrayList<>();
```

Comment:
There should be a comment here on why the list is necessary.

Reply:
Doesn't the name of the field indicate it enough? If the code can be self-documenting, additional comments should not be neated.

Reply:
To me, 'strong holder' doesn't mean anything. 'Strong reference' does, but 'strong holder' does not. Plus, it doesn't hurt to just say 'prevent loaded definitions from being garbage collected'. I won't push on a full blown definition, but something that makes it more obvious that it's making a strong reference to the currently parsed holders.

Reply:
I have renamed the field to indicate better its use.